Core/NoOpRunner.Core/Shapes/ShapeFactories/ImpassableShapeFactory.cs

```diff
@@ -0,0 +1,28 @@
+﻿using NoOpRunner.Core.Enums;
+using NoOpRunner.Core.Shapes.EntityShapes;
+using System;
+
+namespace NoOpRunner.Core.Shapes.ShapeFactories
+{
+    class ImpassableShapeFactory : AbstractFactory
+    {
+        public override EntityShape CreateEntityShape(Shape shape, int x, int y)
+        {
+            switch (shape)
+            {
+                case Shape.Saw: return new Saw(x, y);
+                case Shape.Rocket: return new Rocket(x, y);
+                default: throw new Exception("Shape is not found");
```

Comment:
Try using more specific exception. This should probably be an `ArgumentException`. Same below and in a few other places.

Reply:
Fixed